File: 020-Autobiographical Number/AutobiographicalNumber.py
def FindAutoCount(n):
    if n is None:
        return 0

    count = [0] * 10
    seen = [0] * 10  # To track which digits are already counted as unique
    unique = 0
    length = len(n)

    # Count frequency and prepare unique digit count in one pass
    for i in range(length):
        digit = int(n[i])
        count[digit] += 1
        if seen[digit] == 0:
            unique += 1
            seen[digit] = 1

    # Check autobiographical condition
    for i in range(length):
        if int(n[i]) != count[i]:
            return 0

    return unique

# Example usage
n = "1210"
print(FindAutoCount(n))  # Output: 3

File: 020-Autobiographical Number/AutobiographicalNumber.cpp
#include <iostream>
#include <unordered_set>
#include <vector>
using namespace std;

int FindAutoCount(string n) {
    if (n.empty()) {
        return 0;
    }
    int length = n.length();
    vector<int> count(10, 0); 

    // Count occurrences of each digit
    for (int i = 0; i < length; i++) {
        int digit = n[i] - '0';
        count[digit]++;
    }

    // Validate if the number is autobiographical
    for (int i = 0; i < length; i++) {
        int expected_count = n[i] - '0';
        if (count[i] != expected_count){
            return 0;
        }
            
    }

    // Count unique digits
    unordered_set<char> uniqueDigits;
    for (int i = 0; i < length; i++) {
        uniqueDigits.insert(n[i]);
    }

    return uniqueDigits.size();
}

int main() {
    string n;
    cout << "Enter the number: ";
    cin >> n;
    cout << FindAutoCount(n) << endl;
    return 0;
}